__tests__/app.test.js

```diff
@@ -275,6 +275,75 @@ describe("/api/articles/:article_id/comments", () => {
       });
     });
   });
+
+  describe("POST", () => {
+    describe("Happy paths", () => {
+      test("201: Responds with comment - an object with keys: comment_id, votes, created_at, author, body, article_id", () => {
+        return request(app)
+          .post("/api/articles/5/comments")
+          .send({ username: "rogersop", body: "Hello, world!" })
+          .expect(201)
+          .then(({ body }) => {
+            const comment = body.comment;
+            const [keys, values] = [
+              Object.keys(comment),
+              Object.values(comment),
+            ];
+            expect(keys).toHaveLength(6);
+
+            const validKeys = [
+              "comment_id",
+              "votes",
+              "created_at",
+              "author",
+              "body",
+              "article_id",
+            ];
+            for (let i = 0; i < keys.length; i++) {
```

Comment:
a length check should happen before this, as it's dependent on the length of the keys array :)